erc1155/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+    name: String,
+    symbol: String,
+    description: String,
+    base_uri: String,
+    balances: BTreeMap<u128, BTreeMap<ActorId, u128>>,
+    operator_approvals: BTreeMap<ActorId, BTreeMap<ActorId, bool>>,
+}
+
+static mut ERC1155_TOKEN: Erc1155Token = Erc1155Token {
+    name: String::new(),
+    symbol: String::new(),
+    base_uri: String::new(),
+    description: String::new(),
+    balances: BTreeMap::new(),
+    operator_approvals: BTreeMap::new(),
+};
+
+impl Erc1155Token {
+    fn get_balance(&self, account: &ActorId, id: &u128) -> u128 {
+        *self
+            .balances
+            .get(id)
+            .and_then(|m| m.get(account))
+            .unwrap_or(&0)
+    }
+
+    fn set_balance(&mut self, account: &ActorId, id: &u128, amount: u128) {
+        debug!(
+            "before mint: {:?}, id: {:?}",
+            self.get_balance(account, id),
+            id
+        );
+
+        let mut _balance = self
+            .balances
+            .entry(*id)
+            .or_default()
+            .insert(*account, amount);
+
+        debug!(
+            "after mint: {:?}, id: {:?}",
+            self.get_balance(account, id),
+            id
+        );
+    }
+
+    fn balance_of(&self, account: &ActorId, id: &u128) -> u128 {
+        self.get_balance(account, id)
+    }
+
+    fn balance_of_batch(&self, accounts: &[ActorId], ids: &[u128]) -> Vec<BalanceOfBatchReply> {
+        if accounts.len() != ids.len() {
+            panic!("ERC1155: accounts and ids length mismatch");
+        }
+
+        let mut arr: Vec<BalanceOfBatchReply> = Vec::new();
+
+        for (i, ele) in ids.iter().enumerate() {
+            let account = accounts[i];
+            let amount = self.get_balance(&account, &ele);
+
+            let obj = BalanceOfBatchReply {
+                account: account,
+                id: *ele,
+                amount: amount,
+            };
+
+            arr.push(obj);
+        }
+
+        return arr;
+    }
+
+    fn mint(&mut self, from: &ActorId, id: &u128, amount: u128) {
+        // check owner
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+        let old_balance = self.get_balance(from, id);
+        self.set_balance(from, id, old_balance.saturating_add(amount));
+
+        // TransferSingle event
+    }
+
+    fn mint_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
+        if from == &ZERO_ID {
+            panic!("ERC1155: Mint to zero address");
+        }
+
+        if ids.len() != amounts.len() {
+            panic!("ERC1155: ids and amounts length mismatch");
+        }
+
+        for (i, ele) in ids.iter().enumerate() {
+            let amount = amounts[i];
+            let old_balance = self.get_balance(from, ele);
+            self.set_balance(from, ele, old_balance.saturating_add(amount));
+        }
+
+        // TransferBatch event
+    }
+
+    fn burn_batch(&mut self, from: &ActorId, ids: &[u128], amounts: &[u128]) {
```

Comment:
You must check in that function that account that sends the message (`msg::source()`). Otherwise anyone can burn anyone's tokens